crypto/http/http_lib.c

```diff
@@ -112,14 +138,36 @@ int OSSL_HTTP_parse_url(const char *url, char **phost, char **pport,
         ERR_raise(ERR_LIB_HTTP, HTTP_R_INVALID_URL_PATH);
         goto parse_err;
     }
+    path_end = query = query_end = frag = frag_end = path + strlen(path);
+
+    /* parse optional "?query" */
+    tmp = strchr(path, '?');
+    if (pquery != NULL && tmp != NULL) {
+        path_end = tmp;
+        query = path_end + 1;
+        query_end = frag = frag_end = query + strlen(query);
+    }
+
+    /* parse optional "#fragment" */
+    tmp = strchr(path, '#');
+    if (tmp != NULL) {
+        if (query == path_end) /* we did not record a query component */
+            path_end = tmp;
+        query_end = tmp;
+        frag = query_end + 1;
+        frag_end = frag + strlen(frag);
```

Comment:
Same thing here, `frag_end` is unlikely to get a different value than on line 141...